frontend/src/components/G6Graph.svelte

```diff
@@ -36,127 +36,245 @@
+        edges.forEach((edge) => {
+          edge.toFront(); // put edge on top of the nodes (to see lables)
+          graph.updateItem(edge, {
+            //update the edges of the node (used here to style labels and edge)
+            labelCfg: {
+              style: {
+                fillOpacity: 1, // change the opacity to 1(make it visible), as the default opacity is set to 0(invisible).
+              },
+            },
+            style: {
+              opacity: 1,
+              lineWidth: 1.5,
+            },
+          });
+        });
+      }
+    });
+
+    graph.on("node:mouseleave", (e) => {
+      const node = e.item;
+      if (node instanceof Node) {
+        // check if item is a Node to be able to access the getEdges() method.
+        const edges = node.getEdges();
+        edges.forEach((edge) => {
+          edge.toBack(); // put edge back behond the node
+          graph.updateItem(edge, {
+            labelCfg: {
+              style: {
+                fillOpacity: 0, // make the link lable invisible again, as the mouse moves away from the node
+              },
+            },
+            style: {
+              opacity: 0.15,
+              lineWidth: 1,
+            },
+          });
+        });
+      }
+    });
+
+    // Enable keyboard manipulation
+    graph.on("keydown", (e: IG6GraphEvent) => {
+      let weight: Function = (k1: string, k2: string) =>
+        e.key == k1 ? -1 : e.key == k2 ? 1 : 0;
+      graph.translate(
+        weight("ArrowRight", "ArrowLeft") * graph_translation,
+        weight("ArrowDown", "ArrowUp") * graph_translation
+      );
+    });
+
+    //TODO : Add a listener when entering and exiting the nodes
+    graph.on("node:mouseenter", (e) => dispatch("nodeHovered", e)); // emit an event when the node is entered with the mouse
+
+    graph.changeData(data);
+    resizeGraph();
+
+    return () => {
+      graph.destroy();
+    };
+  });
+
+  $: {
+    if (data && graph) {
+      graph.changeData(data);
     }
+  }
+  reset();
 </script>
 
 <svelte:window on:resize={resizeGraph} />
 
 <div bind:this={container} class="container" />
 
 <style global>
-    .container {
-        height: 100%;
-    }
-    .g6TimeBar {
-        background: #131616;
-        border-radius: 20px;
-        position: absolute !important;
-        left: 35%;
-        bottom: 80px;
-        z-index: 0;
-    }
+  .container {
+    height: 100%;
+  }
+  .g6TimeBar {
+    background: #131616;
+    border-radius: 20px;
+    position: absolute !important;
+    left: 35%;
+    bottom: 80px;
+    z-index: 0;
+  }
+
+  .g6tooltip {
+    background-color: rgb(32, 33, 33, 0.8);
+    border-radius: 8px;
+    align-items: center;
+    border-color: #555555;
+    border-width: 1px;
+    box-shadow: rgb(35, 34, 34) 2px 2px 2px;
+    height: fit-content;
+    align-items: center;
+    width: fit-content;
+    font-size: 0.875rem;
+    color: #ffffff;
+    font-family: "ui-sans-serif", "system-ui", "-apple-system",
```

Comment:
this seems unneeded?

Reply:
Will look it up!

Reply:
Fixed for the next commit (waiting for answers to questions before pushing to have one commit addressing all feedback)

Reply:
So all the fonts are needed?

Reply:
I am not sure about the fonts. I will ask around what was the intention with them =) 
I tested without and I don't see any impact on removing them but since it is not my contribution, I will ask in case there was something I am missing (like some fallback or something).

Reply:
@kam56 Do you have insights regarding the fonts? Can they be removed?

Reply:
I talked to @kam56 and based on the reason she gave me, I removed the fonts.